### ISSUE TITLE
Documentation with 'make' is misleading

### ISSUE THREAD

**Issue by mjwall** [neovim/neovim#16006]
### Neovim version (nvim -v)

0.5.0

### Vim (not Nvim) behaves the same?

not applicable

### Operating system/version

Linux

### Terminal name/version

xterm

### $TERM environment variable

xterm-256-color

### Installation

build from repo

### How to reproduce the issue

Following instructions from https://github.com/neovim/neovim/wiki/Building-Neovim#building
```
git clone git@github.com:neovim/neovim.git
cd neovim/
git checkout stable
make deps
make CMAKE_BUILD_TYPE=Release
make CMAKE_INSTALL_PREFIX=$HOME/opt/nvim-0.5.0 install
```

### Expected behavior

Expected behavior is that nvim will be built at the Release level.  

### Actual behavior

The second make command rebuilds neovim with CMAKE_BUILD_TYPE=Debug which is the default in the makefile, and then installs it.  Passing in CMAKE_BUILD_TYPE to the second make command works as expected, but unlike cmake, make is not caching that value

```
make CMAKE_BUILD_TYPE=Release
make CMAKE_BUILD_TYPE=Release CMAKE_INSTALL_PREFIX=$HOME/opt/nvim-0.5.0 install
```

**Comment by muniter** [comment]
So I think the issue is the `make deps` step, you should not do it since it generates the cache with the default build type that's debug. Try running `make distclean` and try building again with only one command. 

Wiki: 
> CMake, our main build system, caches a lot of things in build/CMakeCache.txt. If you ever want to change CMAKE_BUILD_TYPE or CMAKE_INSTALL_PREFIX, run rm -rf build first. This is also required when rebuilding after a Git commit adds or removes files (including from runtime) — when in doubt, run make distclean (which is basically a shortcut for rm -rf build .deps).

**Comment by mjwall** [comment]
> So I think the issue is the `make deps` step, you should not do it since it generates the cache with the default build type that's debug. Try running `make distclean` and try building again with only one command.
> 
> Wiki:
> 
> > CMake, our main build system, caches a lot of things in build/CMakeCache.txt. If you ever want to change CMAKE_BUILD_TYPE or CMAKE_INSTALL_PREFIX, run rm -rf build first. This is also required when rebuilding after a Git commit adds or removes files (including from runtime) — when in doubt, run make distclean (which is basically a shortcut for rm -rf build .deps).

I don't think that is the issue.  If you don't `make deps`, you can't build unless you manually install all the dependencies.  I agree with what the wiki you quoted states.  

Sorry this is not clear. Let me try again.  Please go to https://github.com/neovim/neovim/wiki/Building-Neovim#building.  The documentation states the following

> So, for a release build, just use:
>
> `make CMAKE_BUILD_TYPE=Release`
>
> Afterwards, the nvim executable can be found in build/bin. To verify the build type after compilation, run:
>
> `./build/bin/nvim --version | grep ^Build`
> 
> To install the executable to a certain location, use:
>
> `make CMAKE_INSTALL_PREFIX=$HOME/local/nvim install`

If you follow those steps, that last one does not install the nvim you just built with CMAKE_BUILD_TYPE=Release.  Instead, it rebuilds nvim with CMAKE_BUILD_TYPE=Debug and then installs that version.  If you start the installed nvim and run `:healthcheck` you will see
```
## Performance
    - OK: Build type: Debug
```

Now for the last step if you instead ran `cmake --build build -D CMAKE_INSTALL_PREFIX=blah install` it would work as expected because then cmake would read CMAKE_BUILD_TYPE from the build/CMakeCache.txt file.  But calling make exactly as quoted in the documentation end up calling `cmake -DCMAKE_BUILD_TYPE=Debug ... ` and overrides what is in the build/CMakeCache.txt file.

I hope that is clearer.  Thanks for looking.

**Comment by clason** [comment]
I'm installing to `/usr/local` by choice, so that's something I haven't tested.

Does 
```
make CMAKE_INSTALL_PREFIX=$HOME/local/nvim install
```
(or `CMAKE... make install`) really not work for you?
